pkg/codegen/java/gen_test.go

```diff
@@ -10,46 +10,100 @@ import (
 	"github.com/pulumi/pulumi/pkg/v3/codegen/testing/test"
 )
 
-var javaSpecificTests = []*test.SDKTest{
-	{
-		Directory:   "mini-azurenative",
-		Description: "Regression tests extracted from trying to codegen azure-natuve",
-	},
-	{
-		Directory:   "mini-awsnative",
-		Description: "Regression tests extracted from trying to codegen aws-native",
-	},
-	{
-		Directory:   "mini-awsclassic",
-		Description: "Regression tests extracted from trying to codegen aws",
-	},
-	{
-		Directory:   "mini-kubernetes",
-		Description: "Regression tests extracted from trying to codegen kubernetes",
-	},
-	{
-		Directory:   "mini-azuread",
-		Description: "Regression tests extracted from trying to codegen azuread",
-	},
-	{
-		Directory:   "mini-awsx",
-		Description: "Regression tests extracted from trying to codegen awsx",
-	},
-	{
-		Directory:   "output-funcs-edgeorder",
-		Description: "Testing EdgeOrder functions which return Output<T>",
-	},
-	{
-		Directory:   "jumbo-resources",
-		Description: "Testing resources with more than 255 properties",
-	},
+// Test case configuration for a GeneratePackage test.
+type generatePackageTestConfig struct {
+	packageInfo     PackageInfo
+	sdkTest         *test.SDKTest
+	hasRuntimeTests bool
 }
 
-func adaptTest(t *test.SDKTest) *test.SDKTest {
+// Configures tests that are maintained in the pulumi-java repository
+// specifially for Java codegen testing.
+func javaSpecificTests() []generatePackageTestConfig {
+	return []generatePackageTestConfig{
+		newGeneratePackageTestConfig(&test.SDKTest{
+			Directory:   "mini-azurenative",
+			Description: "Regression tests extracted from trying to codegen azure-natuve",
+		}),
+		newGeneratePackageTestConfig(&test.SDKTest{
+			Directory:   "mini-awsnative",
+			Description: "Regression tests extracted from trying to codegen aws-native",
+		}),
+		newGeneratePackageTestConfigWithExtras(&test.SDKTest{
+			Directory:   "mini-awsclassic",
+			Description: "Regression tests extracted from trying to codegen aws",
+			// TODO[pulumi/pulumi-java#821]
+			Skip: codegen.NewStringSet("java/test"),
+		}, &PackageInfo{
+			Dependencies: map[string]string{
```

Comment:
just nit, the deps used here are not in line with what we use in sdk

Reply:
Good point. I made a change to sync them to Pulumi's build.gradle file, so that Dependabot PRs upgrade all those together.